Description: Avoid `bytes` type in test process output.
 .
 Direct comparison to objects of type `bytes` can fail when the terminal
 stream is not in the assumed hard-coded encoding.
 .
 Instead, the test cases now avoid direct use of `bytes` where text is
 intended.
Bug-GitHub: https://github.com/twisted/towncrier/issues/479
Author: Ben Finney <bignose@debian.org>
Last-Update: 2023-02-15

diff --git old/src/towncrier/test/test_check.py new/src/towncrier/test/test_check.py
--- old/src/towncrier/test/test_check.py
+++ new/src/towncrier/test/test_check.py
@@ -190,7 +190,7 @@ class TestChecker(TestCase):
             stdout, stderr = proc.communicate()
 
         self.assertEqual(0, proc.returncode)
-        self.assertEqual(b"", stderr)
+        self.assertEqual(0, len(stderr))
 
     def test_first_release(self):
         """
diff --git old/src/towncrier/test/test_project.py new/src/towncrier/test/test_project.py
--- old/src/towncrier/test/test_project.py
+++ new/src/towncrier/test/test_project.py
@@ -190,7 +190,8 @@ class InvocationTests(TestCase):
                 f.write("[tool.towncrier]\n" 'directory = "news"\n')
             os.makedirs("news")
             out = check_output([sys.executable, "-m", "towncrier", "--help"])
-            self.assertIn(b"[OPTIONS] COMMAND [ARGS]...", out)
+            out_text = out.decode(sys.stdout.encoding)
+            self.assertIn("[OPTIONS] COMMAND [ARGS]...", out_text)
             self.assertRegex(out, rb".*--help\s+Show this message and exit.")
         finally:
             os.chdir(orig_dir)
@@ -200,4 +201,5 @@ class InvocationTests(TestCase):
         `--version` command line option is available to show the current production version.
         """
         out = check_output(["towncrier", "--version"])
-        self.assertTrue(out.startswith(b"towncrier, version 2"))
+        out_text = out.decode(sys.stdout.encoding)
+        self.assertTrue(out_text.startswith("towncrier, version 2"))

Local variables:
coding: utf-8
mode: diff
time-stamp-format: "%:y-%02m-%02d"
time-stamp-start: "^Last-Update:[ 	]+"
time-stamp-end: "$"
time-stamp-line-limit: 20
End:
vim: fileencoding=utf-8 filetype=diff :
